static/js/topic_list.js

```diff
@@ -213,37 +237,43 @@ exports.need_to_show_no_more_topics = function (stream_id) {
         return false;
     }
 
-    if (stream_id !== active_widget.get_stream_id()) {
-        return false;
+    if (!active_widgets.has(stream_id)) {
+        return;
```

Comment:
This one should be a `return false`, for consistency of return type.  I changed that before merging.